### PR TITLE
(ROSE-1444) Add rose@0.9.10.0 %gcc@6.3.0 +cxx11

### PR DESCRIPTION
**Pull request by justintoo**
```
$ git clone --branch ROSE-1444-add-rose-0.9.10.0-cxx11 https://github.com/justintoo/spack.git

# Install GCC 6.3.0, which we will add as a Spacker compiler:
$ ./bin/spack install -v gcc@6.3.0
==> '/export/tmp.hudson-rose/rose/ROSE-1387-upgrade-spack-to-rose-0.9.10.0/spack/opt/spack/linux-rhel7-x86_64/gcc-5.1.0/gcc-6.3.0-es2lrwp3daeixb3a4cylby5z5joebjxs/bin/gcc' '-dumpspecs'
==> Successfully installed gcc
  Fetch: 5m 40.44s.  Build: 27m 22.95s.  Total: 33m 3.39s.
[+] /export/tmp.hudson-rose/rose/ROSE-1387-upgrade-spack-t

$ spack compiler add $(spack location --install-dir gcc@6.3.0)
==> Added 1 new compiler to /home/hudson-rose/.spack/linux/compilers.yaml
    gcc@6.3.0
==> Compilers are defined in the following files:
    /home/hudson-rose/.spack/linux/compilers.yaml

$ ./bin/spack install -v rose@0.9.10.0 +cxx11 %gcc@6.3.0
...

$ source share/spack/setup-env.sh

$ spack find rose
==> 1 installed packages.
-- linux-rhel7-x86_64 / gcc@6.3.0 -------------------------------
rose@0.9.10.0

$ spack load rose@0.9.10.0
$ identityTranslator --version
ROSE (version: 0.9.10.0)
  --- using ROSE SCM version:
      --- ID: 66c87e9395126a7ab8663d81f0e0b99d6e09131e
      --- Timestamp: 2017-09-09 02:36:40 UTC
  --- using EDG C/C++ front-end version: edg-4.12
  --- using OFP Fortran parser version: unknown (OFP is disabled)
  --- using Boost version: 1.68.0 (/export/tmp.hudson-rose/rose/ROSE-1387-upgrade-spack-to-rose-0.9.10.0/spack/opt/spack/linux-rhel7-x86_64/gcc-6.3.0/boost-1.68.0-eb55epo66kl3mp4gbrzbiuywz4exqoon)
  --- using backend C compiler: /export/tmp.hudson-rose/rose/ROSE-1387-upgrade-spack-to-rose-0.9.10.0/spack/lib/spack/env/gcc/gcc version: 6.3
  --- using backend C compiler path (as specified at configure time): /export/tmp.hudson-rose/rose/ROSE-1387-upgrade-spack-to-rose-0.9.10.0/spack/lib/spack/env/gcc/gcc
  --- using backend C++ compiler: g++ version: 6.3
  --- using backend C++ compiler path (as specified at configure time): /export/tmp.hudson-rose/rose/ROSE-1387-upgrade-spack-to-rose-0.9.10.0/spack/lib/spack/env/gcc/g++
  --- using original build tree path: /tmp/hudson-rose/spack-stage/spack-stage-_DqdOR/rose-develop-0.9.10.0/rose-build
  --- using instalation path: /export/tmp.hudson-rose/rose/ROSE-1387-upgrade-spack-to-rose-0.9.10.0/spack/opt/spack/linux-rhel7-x86_64/gcc-6.3.0/rose-0.9.10.0-w77qvbcu7qo7d5j6wrxjjh2mncej3lom
  --- using GNU readline version: unknown (readline is disabled)
  --- using libmagic version: unknown (libmagic is disabled)
  --- using yaml-cpp version: unknown (yaml-cpp is disabled)
  --- using lib-yices version: unknown (libyices is disabled)
```

### CODE REVIEW

File: var/spack/repos/builtin/packages/rose/package.py [hunk 1]
```diff
@@ -27,49 +27,128 @@
...
+    version('0.9.9.52', commit='bd4fc0cc332ce62d9fa54db19879507d9e4f239b',
+            git='https://github.com/rose-compiler/rose-develop.git')
+    version('develop', branch='master',
+            git='https://github.com/rose-compiler/rose-develop.git')
 
-    version('master', branch='master')
-    version('0.9.7', 'e14ce5250078df4b09f4f40559d46c75')
+    #--------------------------------------------------------------------------
+    # Dependencies
+    #--------------------------------------------------------------------------
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/rose/package.py]: @justintoo There are several flake errors, most of them are asking you to put a space after any `#`

**Comment by justintoo** [comment]
Added flake8 fixes @scheibelp @tgamblin

**Comment by scheibelp** [comment]
@justintoo there are still a few remaining flake8 complaints

**Comment by justintoo** [comment]
@scheibelp Looks like all the checks have passed. Please let me know if anything further is required to have this PR merged.

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**